### PR TITLE
Created a toggle function that allows users to switch between dark mode and light mode.

### PR DESCRIPTION
**Pull request by Majd-sufian**
Created a toggle function that allows users to switch between dark mode and light mode.

The way it works:

- Open piral browser extension.
- Go to settings that located in the right corner of the page.
- Search for (Switch to dark) or (Switch to light).
- Click on the toggle button and the theme will automatically be switched.
- After toggling the preference theme, it will be saved to local storage for future use.

### CODE REVIEW

**Comment by Majd-sufian** [comment]
Created a toggle function that allows users to switch between dark mode and light mode.

The way it works:

- Open piral browser extension.
- Go to settings that located in the right corner of the page.
- Search for (Switch to dark) or (Switch to light).
- Click on the toggle button and the theme will automatically be switched.

File: src/app/EmitEventModal.tsx [hunk 1]
```diff
@@ -60,7 +61,7 @@ export const EmitEventModal: FC<EmitEventModalProps> = ({ isOpen, toggle }) => {
         </FormGroup>
       </ModalBody>
       <ModalFooter>
-        <Button color="primary" onClick={send} disabled={disabled}>
```
Comment (FlorianRappl) [inline src/app/EmitEventModal.tsx]: Keep primary button.

File: src/app/ExtensionCatalogue.tsx [hunk 2]
```diff
@@ -59,7 +62,7 @@ const ExtensionItem: FC<ExtensionItemProps> = ({ name }) => {
               onChange={setValue}
             />
           </FormGroup>
-          <Button color="primary" onClick={send} disabled={disabled}>
```
Comment (FlorianRappl) [inline src/app/ExtensionCatalogue.tsx]: Keep primary button.

File: src/app/LinkPilets.tsx [hunk 3]
```diff
@@ -17,14 +18,15 @@ export const LinkPilets: FC<LinkPiletsProps> = () => {
     e.preventDefault();
   };
 
+  // detect if the user prefers dark mode
+  const userPrefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+
   return (
-    <Form onSubmit={submit}>
+    <Form onSubmit={submit} css={appSectionView}>
```
Comment (FlorianRappl) [inline src/app/LinkPilets.tsx]: Remove appSectionView.

File: src/app/EmitEventModal.tsx [hunk 4]
```diff
@@ -32,7 +33,7 @@ export const EmitEventModal: FC<EmitEventModalProps> = ({ isOpen, toggle }) => {
   };
 
   return (
-    <Modal isOpen={isOpen} toggle={toggle}>
+    <Modal isOpen={isOpen} toggle={toggle} css={appSectionView}>
```
Comment (FlorianRappl) [inline src/app/EmitEventModal.tsx]: Remove appSectionView.

File: src/app/AdjustSettingsModal.tsx [hunk 5]
```diff
@@ -20,8 +22,13 @@ export const AdjustSettingsModal: FC<AdjustSettingsModalProps> = ({ settings, is
...
   useEffect(() => setValues(initialValues), [initialValues]);
 
+  const [_, api] = store;
+  const { actions } = api.getState();
+
+  const currentTheme = useStore(m => m.state.theme);
+
   return (
-    <Modal isOpen={isOpen} toggle={toggle}>
+    <Modal isOpen={isOpen} toggle={toggle} css={appSectionView}>
```
Comment (FlorianRappl) [inline src/app/AdjustSettingsModal.tsx]: Remove appSectionView.

File: src/app/AdjustSettingsModal.tsx [hunk 6]
```diff
@@ -69,14 +76,18 @@ export const AdjustSettingsModal: FC<AdjustSettingsModalProps> = ({ settings, is
...
+              label={`Switch to ${currentTheme === 'dark' ? 'light' : 'dark'} mode`}
+              id="change theme"
+              onChange={actions.toggleTheme}
+            />
           </div>
         </FormGroup>
         <p style={{ fontSize: '0.8em' }}>We recommend to refresh the page after changing these settings.</p>
       </ModalBody>
       <ModalFooter>
-        <Button color="primary" onClick={send}>
```
Comment (FlorianRappl) [inline src/app/AdjustSettingsModal.tsx]: Keep button as primary.

File: src/app/ExtensionCatalogue.tsx [hunk 7]
```diff
@@ -40,6 +40,9 @@ const ExtensionItem: FC<ExtensionItemProps> = ({ name }) => {
     toggle(ev);
   };
 
+  // detect if the user prefers dark mode
+  const userPrefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (FlorianRappl) [inline src/app/ExtensionCatalogue.tsx]: Remove this

File: src/app/StateContainer.tsx [hunk 8]
```diff
@@ -85,6 +85,7 @@ export const StateContainer: FC<StateContainerProps> = () => {
           <BreadcrumbItem key={`${i}_${item}`}>
             <a
               href="#"
+              className='stateContainer__a'
```
Comment (FlorianRappl) [inline src/app/StateContainer.tsx]: Remove this one

File: src/app/UploadPilets.tsx [hunk 9]
```diff
@@ -36,11 +37,14 @@ export const UploadPilets: FC<UploadPiletsProps> = () => {
...
 
+  // detect if the user prefers dark mode
+  const userPrefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
+
   return (
-    <InputGroup>
+    <InputGroup css={appSectionView}>
       <CustomInput key={file.key} type="file" id="upload-pilet" label="Select a local pilet" onChange={uploadPilet} />
       <InputGroupAddon addonType="append">
-        <Button color="primary" disabled={!file.value} onClick={upload}>
```
Comment (FlorianRappl) [inline src/app/UploadPilets.tsx]: Keep the primary button.

File: src/app/UploadPilets.tsx [hunk 10]
```diff
@@ -36,11 +37,14 @@ export const UploadPilets: FC<UploadPiletsProps> = () => {
     setFile({ value: undefined, key: file.key + 1 });
   };
 
+  // detect if the user prefers dark mode
+  const userPrefersDark = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches;
```
Comment (FlorianRappl) [inline src/app/UploadPilets.tsx]: Remove the old

File: src/app/UploadPilets.tsx [hunk 11]
```diff
@@ -36,11 +37,14 @@ export const UploadPilets: FC<UploadPiletsProps> = () => {
...
+
   return (
-    <InputGroup>
+    <InputGroup css={appSectionView}>
```
Comment (FlorianRappl) [inline src/app/UploadPilets.tsx]: Remove  class

File: src/scripts/legacy/helpers.ts [hunk 12]
```diff
@@ -28,6 +28,19 @@ export function sendVisualizeAll() {
   `);
 }
 
+export function getTheme() {
```
Comment (FlorianRappl) [inline src/scripts/legacy/helpers.ts]: Don't put anything in here.

File: src/app/store.ts [hunk 13]
```diff
@@ -1,8 +1,10 @@
 import create, { SetState } from 'zustand';
 import { PiralDebugCapabilities, PiralDebugSettings, PiralEvent, PiralWorkerInitialState } from '../types';
+import { getTheme } from '../scripts/legacy/helpers';
```
Comment (FlorianRappl) [inline src/app/store.ts]: Don't reference anything from scripts (especially legacy here)

File: src/scripts/legacy/helpers.ts (hunk 12 above)
Comment (FlorianRappl) [reply src/scripts/legacy/helpers.ts]: --> Move this function to utils.ts of the app.

File: src/app/ConnectedView.tsx [hunk 14]
```diff
@@ -7,7 +7,7 @@ import { PiralDebugCapabilities } from '../types';
 import { useViews } from './useViews';
 
 export interface ConnectedViewProps {
-  capabilities: PiralDebugCapabilities;
+  capabilities: PiralDebugCapabilities;
```
Comment (FlorianRappl) [inline src/app/ConnectedView.tsx]: Run prettier in this file.

File: src/app/AdjustSettingsModal.tsx [hunk 15]
```diff
@@ -3,6 +3,8 @@ import { Button, Modal, ModalHeader, ModalBody, ModalFooter, CustomInput, FormGr
 import { jsx } from '@emotion/core';
 import { updateSettings } from './commands';
 import { PiralDebugSettings } from '../types';
+import { appSectionView } from './styles';
```
Comment (FlorianRappl) [inline src/app/AdjustSettingsModal.tsx]: Remove unused import.

File: src/app/EmitEventModal.tsx (hunk 4 above)
Comment (FlorianRappl) [inline src/app/EmitEventModal.tsx]: This still should be removed.

File: src/app/LinkPilets.tsx [hunk 16]
```diff
@@ -2,6 +2,7 @@ import { FC, useState, SyntheticEvent } from 'react';
 import { Button, Form, Input, InputGroup, InputGroupAddon } from 'reactstrap';
 import { jsx } from '@emotion/core';
 import { injectPiletsFromUrl } from './utils';
+import { appSectionView } from './styles';
```
Comment (FlorianRappl) [inline src/app/LinkPilets.tsx]: Remove unused import.

File: src/app/LinkPilets.tsx [hunk 17]
```diff
@@ -22,9 +23,7 @@ export const LinkPilets: FC<LinkPiletsProps> = () => {
       <InputGroup>
         <Input type="text" value={url} onChange={e => setUrl(e.currentTarget.value)} />
         <InputGroupAddon addonType="append">
-          <Button color="primary" disabled={url === ''}>
-            Add
-          </Button>
+          <Button disabled={url === ''}>Add</Button>
```
Comment (FlorianRappl) [inline src/app/LinkPilets.tsx]: Button should be primary.

File: src/app/UploadPilets.tsx [hunk 18]
```diff
@@ -37,7 +38,7 @@ export const UploadPilets: FC<UploadPiletsProps> = () => {
   };
 
   return (
-    <InputGroup>
+    <InputGroup css={appSectionView}>
```
Comment (FlorianRappl) [inline src/app/UploadPilets.tsx]: Remove this one.

File: src/app/View.tsx [hunk 19]
```diff
@@ -17,6 +23,7 @@ export const View: FC<ViewProps> = () => {
         <h2>Not connected</h2>
         <p>You are currently not running a Piral instance in debug mode.</p>
         <p>Note that you need to run Piral v0.10 or later on localhost for Piral Inspector to work.</p>
+        <Button style={{ marginTop: 10 }} onClick={actions.toggleTheme}>Switch to {currentTheme == "dark"? "light": "dark"}</Button>
```
Comment (FlorianRappl) [inline src/app/View.tsx]: Instead of a `style` you could use emotion here - use it with a `css` instruction.

File: src/app/utils.ts [hunk 20]
```diff
@@ -31,4 +31,18 @@ export function checkJson(input: string) {
   } catch {
     return false;
   }
+}
+
+
+export function getTheme() {
+  if (localStorage.getItem('theme')) {
```
Comment (FlorianRappl) [inline src/app/utils.ts]: Don't use this twice.

```
const theme = localStorage.getItem('theme');

if (theme) {
  return theme; 
} else {
...
}
```

File: src/app/EmitEventModal.tsx [hunk 21]
```diff
@@ -3,6 +3,7 @@ import { Button, Modal, ModalHeader, ModalBody, ModalFooter, FormGroup, Label, I
 import { jsx } from '@emotion/core';
 import { emitEvent } from './commands';
 import { checkJson } from './utils';
+import { appSectionView } from './styles';
```
Comment (FlorianRappl) [inline src/app/EmitEventModal.tsx]: Remove unused import.

File: src/app/UploadPilets.tsx [hunk 22]
```diff
@@ -2,6 +2,7 @@ import { FC, ChangeEvent, useState } from 'react';
 import { CustomInput, InputGroup, InputGroupAddon, Button } from 'reactstrap';
 import { jsx } from '@emotion/core';
 import { injectPiletsFromUrl } from './utils';
+import { appSectionView } from './styles';
```
Comment (FlorianRappl) [inline src/app/UploadPilets.tsx]: Remove unused import.

**Review by FlorianRappl — APPROVED** [review]
Looks great! Thanks :beers:

**Pull request merged**